dolphinscheduler-master/src/main/java/org/apache/dolphinscheduler/server/master/utils/DependentExecute.java

```diff
@@ -187,12 +187,6 @@ private DependResult getDependTaskResult(long taskCode, ProcessInstance processI
      */
     private ProcessInstance findLastProcessInterval(Long definitionCode, DateInterval dateInterval) {
 
-        ProcessInstance runningProcess = processService.findLastRunningProcess(definitionCode,
-                dateInterval.getStartTime(), dateInterval.getEndTime());
-        if (runningProcess != null) {
-            return runningProcess;
-        }
-
```

Comment:
I want to know why to remove this code

Reply:
AFAIK, this method is just want to find the latest `processInstance` by `endTime` or `scheduleTime`.
```
find the last one process instance that :
     * 1. manual run and finish between the interval
     * 2. schedule run and schedule time between the interval
     private ProcessInstance findLastProcessInterval(Long definitionCode, DateInterval dateInterval)
```
It doesn't care about the before processInstance.

For example: 
If I have a process instance manual executed 5 times today, and I use this method to query today process instance.
```
11:00 finished
10:00 finished
9:00 running
8:00 finished
7:00 finished
```
This method should return the latest processInstance(11:00), not 9:00.

Reply:
Yep, you're right. There are indeed this bug.